numba/rewrites/static_getitem.py

```diff
@@ -21,19 +21,29 @@ def match(self, interp, block, typemap, calltypes):
                     const = interp.infer_constant(expr.index)
                 except errors.ConstantInferenceError:
                     continue
-                getitems.append((expr, const))
+                getitems[expr] = const
 
         return len(getitems) > 0
 
     def apply(self):
         """
         Rewrite all matching getitems as static_getitems.
         """
-        for expr, const in self.getitems:
```

Comment:
Hmm, the original spelling is much simpler, though. Perhaps `Rewrite.match()` should always be given a block copy? That way, the rewrite can mutate the IR if it likes to...